File: solutions/uri/1047/1047.cpp
#include <iostream>

// https://www.urionlinejudge.com.br/judge/pt/problems/view/1047

struct TimeLen
{
    int hour;
    int minute;
    int timestamp;
};

TimeLen start_time, end_time;

using namespace std;

int get_timestamp(TimeLen &aux_time)
{
    return aux_time.hour * 60 + aux_time.minute;
}

void get_back_hour_min(TimeLen &aux_time)
{
    aux_time.hour = aux_time.timestamp / 60;
    aux_time.minute = aux_time.timestamp % 60;
}

int main()
{
    cin >> start_time.hour >> start_time.minute;
    cin >> end_time.hour >> end_time.minute;

    start_time.timestamp = get_timestamp(start_time);
    end_time.timestamp = get_timestamp(end_time);

    TimeLen result;

    if (start_time.timestamp < end_time.timestamp)
    {
        result.timestamp = end_time.timestamp - start_time.timestamp;
    }
    else if (start_time.timestamp == end_time.timestamp)
    {
        result.timestamp = 24 * 60;
    }
    else
    {
        result.timestamp = end_time.timestamp + 24 * 60 - start_time.timestamp;
    }

    get_back_hour_min(result);

    cout << "O JOGO DUROU " << result.hour << " HORA(S) E " << result.minute << " MINUTO(S)\n";

    return 0;
}

File: solutions/uri/1047/1047.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

# https://www.urionlinejudge.com.br/judge/pt/problems/view/1047

import sys

def is_python3():
    if sys.version_info >= (3, 0):
        return True
    else:
        return False


def timestamp(hour, minute):
    return hour * 60 + minute


def main():
    entrada = input() if is_python3() else raw_input()
    entrada = list(map(int, entrada.split(' ')[:4]))
    start_time = timestamp(*entrada[:2])
    end_time = timestamp(*entrada[2:])

    duration = None

    if start_time < end_time:
        duration = end_time - start_time
    elif start_time == end_time:
        duration = 24*60
    else:
        duration = end_time + 24*60 - start_time

    result = divmod(duration, 60)

    print('O JOGO DUROU', result[0], 'HORA(S) E', result[1], 'MINUTO(S)')


if __name__ == '__main__':
    main()
